gnd/src/test/java/com/google/android/gnd/ui/editobservation/DateFieldViewModelTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.ui.editobservation;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule;
+import androidx.lifecycle.Observer;
+import com.google.android.gnd.model.observation.DateResponse;
+import com.google.android.gnd.model.observation.Response;
+import com.google.android.gnd.persistence.local.LocalDatabaseModule;
+import com.google.android.gnd.rx.Nil;
+import com.google.android.gnd.rx.SchedulersModule;
+import dagger.hilt.android.testing.HiltAndroidTest;
+import dagger.hilt.android.testing.HiltTestApplication;
+import dagger.hilt.android.testing.UninstallModules;
+import io.reactivex.Observable;
+import io.reactivex.observers.TestObserver;
+import java.util.Date;
+import java8.util.Optional;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@HiltAndroidTest
+@UninstallModules({SchedulersModule.class, LocalDatabaseModule.class})
+@Config(application = HiltTestApplication.class)
+@RunWith(RobolectricTestRunner.class)
+public class DateFieldViewModelTest {
+
+  private DateFieldViewModel dateFieldViewModel;
+  //Date represented in milliseconds for date: 2021-09-24T16:40+0000.
+  public static final Date DATE = new Date(1632501600000L);
+
+  @Rule public InstantTaskExecutorRule instantTaskExecutorRule = new InstantTaskExecutorRule();
+
+  @Before
+  public void setup() {
+    dateFieldViewModel = new DateFieldViewModel(null);
+  }
+
+  @Test
+  public void testUpdateResponse() {
+    dateFieldViewModel.updateResponse(DATE);
+    // Observer initialized to read the value of live data.
+    dateFieldViewModel.getResponse().observeForever(new Observer<Optional<Response>>() {
```

Comment:
Still a bit confused; it looks like all this does it remove the observer on change, but it doesn't affect anything else. What am I missing? Same question throughout.

Reply:
https://medium.com/androiddevelopers/unit-testing-livedata-and-other-common-observability-problems-bb477262eb04 here for more reference

Reply:
Ah ok, got it. Can you add a shared util to do this and document well the motivation for it (i.e. we need to observe LiveData in order to kick off cold streams)? Also, for functional interfaces be sure to use lambda notation instead of inline anonymous classes.

Reply:
can you provide more info on this shared util in sharedTest directory or where?.  Using lambda function make this to reference outer class.
https://stackoverflow.com/questions/42479204/value-of-this-in-an-anonymous-class-vs-a-lambda-expression

Reply:
Ah, I see, good point. In that case an anonymous class is fine, provide the code is D.R.Y.  Please see suggested edits in https://github.com/HemanParbhakar/ground-android/pull/8